### PR TITLE
Fix for selectedAll field

### LINKED ISSUE #2221: 'Check all' box is checked when not all options to claim are selected
1. Use this account to test 0x6B73dc97554b78BFdb01bcAf7F0770Fd430bec33
2. Check one investment option to claim, do not select all
3. trace 'Check all' box behavior

**AR**: it appears to be checked

https://watch.screencastify.com/v/ECDm9dmHsD6rD2i4QMwA

**ER**: the box should be checked when ALL options are selected

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #2221

### CODE REVIEW

File: src/custom/pages/Claim/ClaimsTable.tsx [hunk 1]
```diff
@@ -122,9 +124,18 @@ export default function ClaimsTable({
...
+  const { setSelectedAll } = useClaimDispatchers()
 
   const { deployment: start, investmentDeadline, airdropDeadline } = useClaimTimeInfo()
 
+  const paidClaims = getPaidClaims(userClaimData)
+
+  useEffect(() => {
+    if (selected.length === paidClaims.length) {
+      setSelectedAll(true)
+    }
```
Comment (alfetopito) [inline src/custom/pages/Claim/ClaimsTable.tsx:136]: Shouldn't it be updated when it's not fully selected as well?

```suggestion
  setSelectedAll(selected.length === paidClaims.length)
```

Comment (nenadV91) [reply src/custom/pages/Claim/ClaimsTable.tsx:136]: No because if you unselect one option it will set selectedAll to false.

Comment (W3stside) [reply src/custom/pages/Claim/ClaimsTable.tsx:136]: ...which is what we want 😂

Comment (nenadV91) [reply src/custom/pages/Claim/ClaimsTable.tsx:136]: Yes

**Review by elena-zh — APPROVED** [review]
Changes LGTM!

**Review by anxolin — COMMENTED** [review]
The PR works as expected, although I see a bit of a mess on the responsbilities of the components as discussed in https://gnosisinc.slack.com/archives/C025G521XQD/p1642676703117800?thread_ts=1642674739.113100&cid=C025G521XQD

**Pull request merged**